src/main/java/com/stripe/model/BalanceTransaction.java

```diff
@@ -146,7 +148,16 @@ public String getType() {
 	public void setType(String type) {
 		this.type = type;
 	}
+	
+	public void setSourceObject(HasId sourceObject) {
+		this.sourceObject = sourceObject;
+	}
 
+	@SuppressWarnings("unchecked")
+	public <O> O getSourceObject() {
+		return (O) sourceObject;
+	}
```

Comment:
Hi @brandur-stripe, 

So, I went ahead and double-checked it with the code changes you provided above.

It still fails with the same exception upon the following call:

```java
Event event = APIResource.GSON.fromJson(json, Event.class);
```

(where `json` is JSON structure coming in via webhook)

It throws the following exception:

```
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.balance_transactions[0].source
```

It seems like stripe-java deserialization works in a way that would not handle objects within array (in this case - `balance_transactions` array property) same way non-array properties are handled.

Here’s JSON structure coming in via webhook that it fails to deserialize:

```json
{
  "id": "evt_1AF5AmKac2qaDcCZULVeSfjz",
  "object": "event",
  "api_version": "2017-02-14",
  "created": 1493747548,
  "data": {
    "object": {
      "id": "dp_1AF5AmKac2qaDcCZHvOjyrDo",
      "object": "dispute",
      "amount": 5000,
      "balance_transaction": "txn_1AF5AmKac2qaDcCZm8tBQt6I",
      "balance_transactions": [
        {
          "id": "txn_1AF5AmKac2qaDcCZm8tBQt6I",
          "object": "balance_transaction",
          "amount": -5000,
          "available_on": 1494288000,
          "created": 1493747548,
          "currency": "usd",
          "description": "Chargeback withdrawal for ch_1AF5AlKac2qaDcCZ4Iu07uIA",
          "fee": 1500,
          "fee_details": [
            {
              "amount": 1500,
              "application": null,
              "currency": "usd",
              "description": "Dispute fee",
              "type": "stripe_fee"
            }
          ],
          "net": -6500,
          "source": "dp_1AF5AmKac2qaDcCZHvOjyrDo",
          "status": "pending",
          "type": "adjustment"
        }
      ],
      "charge": "ch_1AF5AlKac2qaDcCZ4Iu07uIA",
      "created": 1493747548,
      "currency": "usd",
      "evidence": {
        "access_activity_log": null,
        "billing_address": "1400 Key Blvd #102, Arlington, VA, 22209, US",
        "cancellation_policy": null,
        "cancellation_policy_disclosure": null,
        "cancellation_rebuttal": null,
        "customer_communication": null,
        "customer_email_address": "asdf@asdfasdfasdfasdfasdf.com",
        "customer_name": "First Last",
        "customer_purchase_ip": "111.111.111.111",
        "customer_signature": null,
        "duplicate_charge_documentation": null,
        "duplicate_charge_explanation": null,
        "duplicate_charge_id": null,
        "product_description": null,
        "receipt": null,
        "refund_policy": null,
        "refund_policy_disclosure": null,
        "refund_refusal_explanation": null,
        "service_date": null,
        "service_documentation": null,
        "shipping_address": null,
        "shipping_carrier": null,
        "shipping_date": null,
        "shipping_documentation": null,
        "shipping_tracking_number": null,
        "uncategorized_file": null,
        "uncategorized_text": null
      },
      "evidence_details": {
        "due_by": 1495151999,
        "has_evidence": false,
        "past_due": false,
        "submission_count": 0
      },
      "is_charge_refundable": false,
      "livemode": false,
      "metadata": {
      },
      "reason": "general",
      "status": "needs_response"
    }
  },
  "livemode": false,
  "pending_webhooks": 1,
  "request": "req_AaFg0ircrTmUJ5",
  "type": "charge.dispute.created"
}
```

I don’t know why there’s both `balance_transaction` (String ID) and `balance_transactions` (array which contains that only balance transaction, but also includes the rest of the transaction properties) within data coming in via webhook, but in any case - it fails while trying to process the very first element of that array.